rust/src/lib/nm/query_apply/apply.rs

```diff
@@ -72,6 +72,8 @@ pub(crate) async fn nm_apply(
         }
     }
 
+    let forwarding_supported = is_forwarding_supported(&nm_api).await;
```

Comment:
You forcing every user to check whether forwarding is supported or not even they are not using this feature.

Instead of check before apply, you should wrap failure error when `connection_add()` for forwarding not supported, so that other user will never been impacted.

Reply:
I dropped the function `is_forwarding_supported()` and changed it as you suggested, thanks!